src/herder/TransactionQueue.cpp

```diff
@@ -127,9 +127,15 @@ findBySeq(int64_t seq, TransactionQueue::TimestampedTransactions& transactions,
         return false;
     }
 
-    releaseAssert(seq - firstSeq <= static_cast<int64_t>(transactions.size()));
-    iter = transactions.begin() + (seq - firstSeq);
-    releaseAssert(iter == transactions.end() || iter->mTx->getSeqNum() == seq);
+    iter = std::lower_bound(transactions.begin(), transactions.end(), seq,
```

Comment:
condition above is wrong: it is now valid to add a tx with seqnum much larger than lastSeq (as long as `minSeqNum` is set)

Reply:
this points to gaps in tests

Reply:
I wrote a test that added a gap and it worked, but I think I see where I had a misunderstanding. `findBySeq` is supposed to return true even if the tx you search for is missing **if** it's the next one in line. I think this only matters for the fee bump case, which I haven't tested yet. Good catch.